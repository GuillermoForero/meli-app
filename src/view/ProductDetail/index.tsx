import Box from 'components/Box';
import React from 'react';
import Header from 'components/Header';
import SearchResult from 'components/SearchResult';
import { COLORS } from 'consts';

const ProductDetail: React.FC = () => (
  <Box alignItems="center" flexDirection="column" style={{ backgroundColor: `${COLORS.DEFAULT_GRAY}` }} minHeight="100vh">
    <Header/>
    <SearchResult/>
  </Box>
  
);

export default ProductDetail;