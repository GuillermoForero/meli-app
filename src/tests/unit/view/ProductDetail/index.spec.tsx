import React from 'react';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import { render } from 'tests/support/customRender';
import { appStoreFull } from 'tests/support/stubAppStore';
import ProductDetail from 'view/ProductDetail';

describe('[ProductDetail]', () => {
  const mockStore = createMockStore();
  it('Renders default snapshot', () => {
    const store = mockStore(appStoreFull);
    const { container, getByText } = render(
      <Provider store={store}>
        <ProductDetail />
      </Provider>);
    expect(getByText('Descripci√≥n del producto')).toBeDefined();
    expect(getByText('title_1')).toBeDefined();
    expect(getByText('description_1')).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
