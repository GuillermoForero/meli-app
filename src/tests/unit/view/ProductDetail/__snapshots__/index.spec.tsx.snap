// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ProductDetail] Renders default snapshot 1`] = `
<div>
  <div
    class="sc-bdfBwQ iBvJIy"
    width="100%"
  >
    <div
      class="ant-row"
      style="width: 100%;"
    >
      <div
        class="ant-col ant-col-2"
      />
      <div
        class="ant-col ant-col-20"
        style="background-color: white;"
      >
        <div
          class="sc-bdfBwQ hONsNi"
        >
          <div
            class="sc-bdfBwQ fOdiPt"
            width="70%"
          >
            <img
              alt="ItemImg"
              class="sc-hKgILt cMdvAc"
              src="url"
            />
            <h4
              class="sc-gsTCUz dWZGgs"
              color="GRAY_20"
              font-size="28px"
            >
              Descripci√≥n del producto
            </h4>
            <p
              class="sc-gsTCUz eugcRc"
              color="GRAY_60"
              font-size="16px"
            >
              description_1
            </p>
          </div>
          <div
            class="sc-bdfBwQ eezWHV"
            width="30%"
          >
            <div
              class="sc-bdfBwQ itqDgf"
            >
              <span
                class="sc-gsTCUz hPYUwp"
                color="GRAY_40"
              >
                new
                 - 234 vendidos
              </span>
            </div>
            <h2
              class="sc-gsTCUz HgnEE"
              color="GRAY_20"
              font-size="24px"
            >
              title_1
            </h2>
            <p
              class="sc-gsTCUz iTjGZM"
              color="GRAY_20"
              font-size="46px"
            >
              $ 
              price_1
            </p>
            <button
              class="sc-dlfnbm jxOxHE"
            >
              Comprar
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
